purchase_order_approved/tests/test_purchase_order.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields
+from odoo.tests.common import TransactionCase
+
+
+class TestPurchaseOrder(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestPurchaseOrder, cls).setUpClass()
+        cls.supplier = cls.env.ref("base.res_partner_1")
+        vals = {
+            "name": "PO TEST",
+            "partner_id": cls.supplier.id,
+        }
+        cls.purchase_order = cls.env["purchase.order"].create(vals)
+        product = cls.env.ref("product.product_product_4")
+        cls.env["purchase.order.line"].create(
+            {
+                "order_id": cls.purchase_order.id,
+                "product_id": product.id,
+                "date_planned": fields.Datetime.now(),
+                "name": "Test",
+                "product_qty": 10.0,
+                "product_uom": product.uom_id.id,
+                "price_unit": 100.0,
+            }
+        )
+        cls.env.invalidate_all()
```

Comment:
- Are you sure this is necessary? What happens if this is not included?
- If so, are you sure all needs to be invalidated?? (i.e. not enough refreshing the record or model)

Reply:
- Yes, it is part of [tasks to do for migration](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0#tasks-to-do-in-the-migration) because the method refresh is deprecated.
- If you watch this [method in 16.0](https://github.com/odoo/odoo/blob/3bd28525d9486e2f33b8d338c39675125fa22d7a/odoo/models.py#L5884), they use env.invalidate_all after the warning.

Reply:
I think I didn't explain myself.

I mean, was this really necessary before? I ask because, if not, we can remove it.

On the other hand, I know before all cache was invalidated. That's why the method was split into small invalidation choices, to prevent the whole cache from being invalidated on every case, because in most of them, only a specific record or model needed invalidation.

Reply:
it can be removed.